test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.cs

```diff
@@ -148,6 +154,65 @@ public async Task CheckEnrichmentWhenSampling()
         await CheckEnrichment(new AlwaysOnSampler(), true, this.url).ConfigureAwait(false);
     }
 
+#if NET8_0_OR_GREATER
+    [Theory]
+    [MemberData(nameof(TestData))]
+    public async Task ValidateNet8MetricsAsync(HttpTestData.HttpOutTestCase tc)
+    {
+        var metrics = new List<Metric>();
+        var meterProvider = Sdk.CreateMeterProviderBuilder()
+            .AddHttpClientInstrumentation()
+            .AddInMemoryExporter(metrics)
+            .Build();
+
+        var testUrl = HttpTestData.NormalizeValues(tc.Url, this.host, this.port);
+
+        try
+        {
+            using var c = new HttpClient();
+            using var request = new HttpRequestMessage
+            {
+                RequestUri = new Uri(testUrl),
+                Method = new HttpMethod(tc.Method),
+            };
+
+            request.Headers.Add("contextRequired", "false");
+            request.Headers.Add("responseCode", (tc.ResponseCode == 0 ? 200 : tc.ResponseCode).ToString());
+            await c.SendAsync(request).ConfigureAwait(false);
+        }
+        catch (Exception)
+        {
+            // test case can intentionally send request that will result in exception
+        }
+        finally
+        {
+            meterProvider.Dispose();
+        }
+
+        // dns.lookups.duration is a typo
+        // https://github.com/dotnet/runtime/issues/92917
+        var requestMetrics = metrics
```

Comment:
nit: Use `Count()` instead of `Where()`

Reply:
nit: A stronger test would be to individually look for each expected metric instead of looking at the aggregated count.

Reply:
good point. added for follow up here https://github.com/open-telemetry/opentelemetry-dotnet/issues/4972